diff --git a/dashboard.el b/dashboard.el
index 808edf9..42c2960 100644
--- a/dashboard.el
+++ b/dashboard.el
@@ -467,7 +467,7 @@ Optional argument ARGS adviced function arguments."
   "Setup post initialization hooks.
 If a command line argument is provided, assume a filename and skip displaying
 Dashboard."
-  (when (< (length command-line-args) 2)
+  (when (not command-line-args-left)
     (add-hook 'window-setup-hook (lambda ()
                                    ;; 100 means `dashboard-resize-on-hook' will run last
                                    (add-hook 'window-size-change-functions 'dashboard-resize-on-hook 100)
